pyroomacoustics/libroom_src/room.cpp

```diff
@@ -276,3 +276,422 @@ int Room::image_source_shoebox(
+	// What we need to trace the ray
+	float energy = 1000;
+	VectorXf start = source_pos;
+	VectorXf end = compute_segment_end(start, max_dist, phi, theta);
+	
+	// The following initializations are arbitrary and does not count since we set
+	// the boolean to false
+	Wall wall = walls[0];
+	Vector1i next_wall_index(0);
+	bool there_is_prev_wall = false;
+	
+	// The ray's characteristics
+	float travel_time = 0;
+	float total_dist = 0;
+	//---------------------------------------------
+	
+	//------------------ RAY TRACING --------------------
+	
+	while(true){
+		
+		
+		/*//Debugging
+		std::cout << "---\n" << "start : " << start[0] << " " << start[1] << " " << start[2] << std::endl;
+		std::cout << "end : " << end[0] << " " << end[1] << " " << end[2] << std::endl;
+		std::cout << "there_is_prev_wall : " <<  there_is_prev_wall <<std::endl;
+		std::cout << "next_wall_index : " <<  next_wall_index[0] <<std::endl;
+		*/
+		
+		
+		VectorXf hit_point = next_wall_hit(start,
+										   end,
+										   there_is_prev_wall,
+										   wall,
+										   next_wall_index);
+
+
+		// Debugging
+		//std::cout << "hit_point : " << hit_point[0]<< " " << hit_point[1]<< " " << hit_point[2] << std::endl;
+		
+		// The wall that has just been hit
+		wall = walls[next_wall_index[0]];
+		there_is_prev_wall = true;
+		
+		// Initialization needed for the next if
+		float distance(0);
+		
+		
+		
+		
+		// Before updatign the ray's characteristic, we must see
+		// if the mic is intersected on the [start, hit_point] segment
+		
+		// - If yes, we compute the energy at the mic_intersection point
+		// and we continue the ray
+		if (intersects_mic(start, hit_point, mic_pos, mic_radius)){
+			
+			Eigen::VectorXf hit_point_mic = mic_intersection(start, hit_point, mic_pos, mic_radius);
+			
+			//std::cout << "MICROPHONE hit_point : " << hit_point_mic[0] << " "<< hit_point_mic[1] << " "<< hit_point_mic[2] << std::endl;
+			distance = (start - hit_point_mic).norm();
+			
+			float travel_time_to_mic = travel_time;
+
+			update_travel_time(travel_time_to_mic, distance, sound_speed);
+			
+			if (travel_time_to_mic < time_thres){
+				append(travel_time_to_mic, energy, output);
+			}
+			
+		}
+		
+		
+		// Update the characteristics
+		
+		//std::cout << "\n\n-------\nprev travel time : " <<  travel_time <<std::endl;
+		//std::cout << "prev energy : " <<  energy <<std::endl;
+		distance = (start - hit_point).norm();
+		total_dist += distance;
+		update_travel_time(travel_time, distance, sound_speed);
+		update_energy_wall(energy, wall);
+		
+		//std::cout << "post travel time : " <<  travel_time <<std::endl;
+		//std::cout << "post energy : " <<  energy <<std::endl;
+		
+		// Check if we reach the time threshold for this ray
+		if (travel_time > time_thres){
+			break;
+		}
+		
+		// Let's simulate the scattered ray induced by the rebound on
+		// the wall
+		
+		if (scatter_coef > 0){
+			
+			float ray_scat_energy = compute_scat_energy(energy,
```

Comment:
Are you talking about the indentation of the parameters (which I tried to align with the first one) ? Or are you talking about the overall indentation ? 

I have the habit to indent everything using 1 tab

Reply:
I see. The project is using 2 spaces for the cpp code. Four spaces in python. I am sorry to go against what you usually do but it might be better to keep the indentation consistent. This can probably also fixed automatically in some way.